rviz_default_plugins/src/rviz_default_plugins/displays/marker/markers/triangle_list_marker.cpp

```diff
@@ -55,6 +62,39 @@ namespace displays
 namespace markers
 {
 
+resource_retriever::MemoryResource getResource(const std::string & resource_path)
+{
+  resource_retriever::Retriever retriever;
+  resource_retriever::MemoryResource res;
+  try {
+    res = retriever.get(resource_path);
+  } catch (resource_retriever::Exception & e) {
+    RVIZ_COMMON_LOG_DEBUG(e.what());
+    return resource_retriever::MemoryResource();
+  }
+
+  return res;
+}
+
+bool loadImage(const Ogre::String & texture_name, const Ogre::String & texture_path)
+{
+  bool image_loaded = false;
+  resource_retriever::MemoryResource res = getResource(texture_path);
```

Comment:
If `getResource` fails, we're passing a nullptr to `Ogre::MemoryDataStream` below; I'm not sure what the consequences of this are. I think it would be safer to guard against this and return early.

Reply:
Not using resource retriever anymore. Currently supporting embedded compressed images.